File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.java
class Solution {
    public int numberOfBeams(String[] bank) {
        int prev=0, ans=0;
        
        for(String s:bank){
            int occ=0;
            for(int i=0;i<s.length();i++){
                if(s.charAt(i)=='1'){
                    occ++;
                }
            }
            if(occ!=0){
                ans +=(prev*occ);
                prev=occ;
            }
        }
        return ans;

    }
}

File: 2244-number-of-laser-beams-in-a-bank/number-of-laser-beams-in-a-bank.cpp
class Solution {
public:
    int numberOfBeams(vector<string>& bank) {
        int ans=0,prev=0;
        
        for(string s:bank){
            int occ=0;
            for(char c:s){
                if(c=='1'){
                    occ++;
                }
            }
            if(occ !=0){
                ans +=(prev*occ);
                prev=occ;
            }
        }
        return ans;
    }
};